File: src/RentCars.Domain/Infrastructure/AuthResponse.cs
namespace RentCars.Domain.Infrastructure;

public class AuthResponse
{
    public String Token { get; }
    public Boolean IsAdmin { get; }
    public String UserId { get; set; }
    public String UserName { get; set; }

    public AuthResponse(String token, Boolean isAdmin, String userId, String userName)
    {
        Token = token;
        IsAdmin = isAdmin;
        UserId = userId;
        UserName = userName;
    }
}

File: API/api.client/src/domain/infrastructure/authResponse.ts
export class AuthResponse {
    constructor(
        public token: string,
        public isAdmin: boolean,
        public userId: string,
        public userName: string
    ) { }
}

export function mapToAuthResponse(data: any): AuthResponse {
    return new AuthResponse(data.token, data.isAdmin, data.userId, data.userName)
}